decidim-core/app/views/layouts/decidim/_wrapper.html.erb

```diff
@@ -20,7 +20,7 @@ end
       <%= render partial: "layouts/decidim/header/main" %>
     </div>
     <div id="menu-bar-container">
-      <% unless current_user&.ephemeral? %>
+      <% if !current_user&.ephemeral? && !content_for?(:focus_mode) %>
```

Comment:
From what i understand, we want to hide the header when user is ephemeral AND there is no focus_mode.. Maybe we could use : 

```suggestion
      <% unless current_user&.ephemeral? && content_for?(:focus_mode) %>
```

Also, i would rename that "focus_mode" to something else like "(show|hide)_menu"